net/smartdns/Makefile

```diff
@@ -0,0 +1,58 @@
+#
+# Copyright (c) 2018-2019 Nick Peng (pymumu@gmail.com)
+# This is free software, licensed under the GNU General Public License v3.
+#
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=smartdns
+PKG_VERSION:=1.2019.28
+PKG_RELEASE:=1
+
+PKG_SOURCE_PROTO:=git
+PKG_SOURCE_URL:=https://www.github.com/pymumu/smartdns.git
+PKG_SOURCE_VERSION:=cc93156ee3af9de89e8e6db72554a3790170ce0b
+PKG_MIRROR_HASH:=89199413a3860d8245da6e4c6f171d449b89ec56162579004fa2166a29d20159
+
+PKG_MAINTAINER:=Nick Peng <pymumu@gmail.com>
+PKG_LICENSE:=GPL-3.0-or-later
+
+PKG_BUILD_PARALLEL:=1
+
+include $(INCLUDE_DIR)/package.mk
+
+MAKE_ARGS:=CFLAGS="$(TARGET_CFLAGS) -DSMARTDNS_VERION=\\\"$(PKG_VERSION)\\\"" CC=$(TARGET_CC) 
+MAKE_PATH:=src
+
+define Package/smartdns
+  SECTION:=net
+  CATEGORY:=Network
+  TITLE:=smartdns server
+  DEPENDS:=+libopenssl
+  URL:=http://www.github.com/pymumu/smartdns/
+endef
+
+define Package/smartdns/description
+SmartDNS is a local DNS server which accepts DNS query requests from local network clients,
+get DNS query results from multiple upstream DNS servers concurrently, and returns the fastest IP to clients.
+Unlike dnsmasq's all-servers, smartdns returns the fastest IP. 
+endef
+
+define Package/smartdns/conffiles
+/etc/config/smartdns
+/etc/smartdns/address.conf
+/etc/smartdns/blacklist-ip.conf
+/etc/smartdns/custom.conf
+endef
+
+define Package/smartdns/install
+	$(INSTALL_DIR) $(1)/usr/sbin $(1)/etc/config $(1)/etc/init.d $(1)/etc/smartdns
+	$(INSTALL_BIN) $(PKG_BUILD_DIR)/src/smartdns $(1)/usr/sbin/smartdns
+	$(INSTALL_BIN) $(PKG_BUILD_DIR)/package/openwrt/files/etc/init.d/smartdns $(1)/etc/init.d/smartdns
```

Comment:
I took a look at this. I recommend updating it to use procd as well as running it through shellcheck.

Reply:
Is there any example?

Reply:
grep the tree for USE_PROCD

Reply:
OK， I think smartdns already use that feature, please help to check
https://github.com/pymumu/smartdns/blob/master/package/openwrt/files/etc/init.d/smartdns#20

Reply:
Please also run it through shellcheck. There are some bashisms in there.

Reply:
done，please review.

Reply:
I see tons of warnings still.

Reply:
I submitted a PR fixing several of those warnings.